File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)

        pre = []
        post = []

        pre.append(1)
        multiplier = 1
        for i in range(0, n-1):
            multiplier *= nums[i]
            pre.append(multiplier)
        
        multiplier = 1
        post.append(1)
        for i in range(n-1, 0, -1):
            multiplier *= nums[i]
            post.append(multiplier)
        post.reverse()

        print(pre)
        print(post)

        res = [0 for i in range(0, n)]
        for i in range(0, n):
            res[i] = pre[i]*post[i]
        

        return res

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int cnt = 1, idx = 1;

        for(int i = 1 ; i < n ; i++){
            if(nums[i] != nums[i-1]){
                nums[idx] = nums[i];
                idx++;
                cnt++;
            }
        }
        return cnt;
    }
};